account_parent/models/account.py

```diff
@@ -0,0 +1,127 @@
+        if context.get('date_to'):
+            res += " AND l.{} <= '{}' ".format(date_field, context['date_to'])
+        if context.get('date_from'):
+            if not context.get('strict_range'):
+                res += " AND ({} >= '{}' ".format(
+                    date_field, context['date_from'], )
+                res += " OR at.include_initial_balance = true) "
+            elif context.get('initial_bal'):
+                res += " AND l.{} < '{}' ".format(
+                    date_field, context['date_from'])
+            else:
+                res += " AND l.{} >= '{}' ".format(
+                    date_field, context['date_from'])
+        if context.get('journal_ids'):
+            res += " AND l.journal_id in {} ".format(
+                tuple(context['journal_ids']))
+        state = context.get('state')
+        if state and state.lower() != 'all':
+            res += """ AND m.state = '%s' """ % state
+        if context.get('company_id'):
+            res += " AND l.company_id = {} ".format(context['company_id'])
+        if 'company_ids' in context:
+            res += " AND l.company_id  in {} ".format(
+                tuple(context['company_ids']))
+        query += res and res
+        return query
+
+    @api.multi
+    @api.depends('move_line_ids', 'move_line_ids.amount_currency',
+                 'move_line_ids.debit', 'move_line_ids.credit')
+    def _compute_values(self):
+        for account in self:
+            query = account._move_domain_get()
+            sub_accounts = self.with_context(
+                {'show_parent_account': True}).search([
+                    ('id', 'child_of', [account.id])])
+            self.env.cr.execute(query, (tuple(sub_accounts.ids),))
+            result = self.env.cr.dictfetchall()[0]
+            account.balance = result['balance']
+            account.credit = result['credit']
+            account.debit = result['debit']
+
+    move_line_ids = fields.One2many(
+        'account.move.line', 'account_id', 'Journal Entry Lines')
+    balance = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    credit = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    debit = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    parent_id = fields.Many2one(
+        'account.account', 'Parent Account', ondelete='restrict')
+    child_ids = fields.One2many(
+        'account.account', 'parent_id', 'Child Accounts')
+    parent_left = fields.Integer('Left Parent', index=1)
+    parent_right = fields.Integer('Right Parent', index=1)
+
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = 'code, name'
+    _order = 'parent_left'
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        context = self._context or {}
+        if not context.get('show_parent_account', False):
```

Comment:
I think it's make better if we check `show_parent_account` is True. May be, we will change many code.
Because another line in all module don't has set this context before. If we check fail, all line will go into condition
`if not context.get('show_parent_account', False):` > `if context.get('show_parent_account'):`